tests/framework/crossplane.go

```diff
@@ -0,0 +1,218 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+// ExpectedNginxField contains an nginx directive key and value,
+// and the expected file, server, and location block that it should exist in.
+type ExpectedNginxField struct {
+	// Key is the directive name.
+	Key string
+	// Value is the value for the directive. Can be the full value or a substring.
+	Value string
+	// File is the file name that should contain the directive. Can be a full filename or a substring.
+	File string
+	// Location is the location name that the directive should exist in.
+	Location string
+	// Servers are the server names that the directive should exist in.
+	Servers []string
+	// ValueSubstringAllowed allows the expected value to be a substring of the real value.
+	// This makes it easier for cases when real values are complex file names or contain things we
+	// don't care about, and we just want to check if a substring exists.
+	ValueSubstringAllowed bool
+}
+
+// ValidateNginxFieldExists accepts the nginx config and the configuration for the expected field,
+// and returns whether or not that field exists where it should.
+func ValidateNginxFieldExists(conf *Payload, expFieldCfg ExpectedNginxField) bool {
```

Comment:
What's the error message look like if this fails? Will we have enough information to debug? It seems like we are losing information by returning a bool instead of an error message.

Reply:
```
failureMsg := fmt.Sprintf(
	"directive '%s' with value '%s' not found in expected place",
	expCfg.Directive, expCfg.Value,
)
```

Was what I came up with to help diagnose this a bit. Maybe there's more context that could be added.

Reply:
I think it would be helpful to distinguish between file-related errors, such as "file doesn't exist," and directive-level errors. Honestly, for debugging, we probably want to dump the entire JSON payload.

Reply:
Updated to spit out the nginx conf. It's messy but can be put into a json formatter for easier reading.